.github/workflows/test.yml

```diff
@@ -82,13 +84,80 @@ jobs:
+          } | Set-Content "win_Dockerfile"
+          docker-compose -f win_docker-compose.yaml up -d --build
+
+          # retrieve id_rsa file for public key authentication
+          mkdir $env:USERPROFILE/.ssh/
+          docker run ipyparallel-sshd powershell.exe -Command "type C:\Users\ciuser\.ssh\id_rsa" | out-file -encoding ascii $env:USERPROFILE/.ssh/id_rsa
+
+          # install newer version of openssh (removes "GetConsoleMode on STD_INPUT_HANDLE failed with 6" error in pytest)
+          # also disable download progress to keep github runner output more compact
+          choco install openssh --pre --no-progress
+          mv C:\Windows\System32\OpenSSH C:\Windows\System32\_OpenSSH
+
+          #check ssh connection and accept host key (with arbitrary windows command)
+          ssh -o "StrictHostKeyChecking no" $env:SSH_HOST -p $env:SSH_PORT -v echo "ssh connection to container succeeded"
+
+          # copy ipyparallel code to docker container (use zip, scp and unzip)
+          ssh $env:SSH_HOST -p $env:SSH_PORT mkdir $env:CODE_ROOT
+
+          # zip ipyparallel files (excluding files probably not needed)
+          cd ../..
+          $exclude = @("__pycache__", "node_modules", ".yarn")
+          $files = Get-ChildItem -Path "." -Exclude $exclude
+          Compress-Archive -Path $files -DestinationPath ipyparallel.zip -CompressionLevel Fastest
+          # copy file into docker (we need to do it over ssh since docker copy or mount doesn't work in Hyper-V)
+          scp -P $env:SSH_PORT ipyparallel.zip ${env:SSH_HOST}:${env:CODE_ROOT}
+          # deflate ipyparallel files
+          ssh $env:SSH_HOST -p $env:SSH_PORT powershell.exe -Command "Expand-Archive -Path $env:CODE_ROOT\ipyparallel.zip -DestinationPath $env:CODE_ROOT"
+
+          # pip install ipyparallel files
+          #ssh $env:SSH_HOST -p $env:SSH_PORT "echo IPP_DISABLE_JS=$env:IPP_DISABLE_JS"
+          ssh $env:SSH_HOST -p $env:SSH_PORT "pip install -e file://c:/src/ipyparallel#egg=ipyparallel[test]"
+
+          # we need to disable the windows firewall for github runners otherwise the ipyparallel engines cannot connect to the controller.
+          # obviously, a more precautious adaption of the firewall would be desirable. since an adaption of the firewall is NOT necessary
+          # for a local standard windows environment, no further improvements were made.
+          echo "Disable Firewall:"
+          Set-NetFirewallProfile -Profile Domain, Public, Private -Enabled False
+
+          # just see were pip is installed and what libraries are install 
+          echo "Check pip inside container"
+          ssh $env:SSH_HOST -p $env:SSH_PORT "where pip"
+          ssh $env:SSH_HOST -p $env:SSH_PORT "pip list"
+
+          echo "Check if container can ping the docker host (requires adapted hosts file and firewall disabled)"
+          docker run ipyparallel-sshd ping -n 1 $env:computername
```

Comment:
Since this test is so different for Windows, maybe just the new `test-ssh.yml` should test ssh instead of mixing it in here. That might remove a lot of complicated conditionals.

Reply:
I do not have a strong opinion on this topic. It reduces the complexity a little bit on one side, but duplicates some parts on the other side. The ssh test under Windows takes much longer than all the other tests, which could be a small argument for splitting the tests. Although it would not run faster, this test could be deactivated for some commits. Would this suite your coding workflows?
I did split up a ssh-test in my fork, but only to have a 'small' test to the bare minimum.

Tell me what you perfer and I will do the necessary changes...

Reply:
Let's go with splitting. I think a lot of things are likely to be different, and it will probably be easier to manage if it's separate.

Reply:
All ssh tests are now done in a separat action file (`test-ssh.yml`). So `test.yml` is back to the original state but also the linux ssh parts removed. There are two things to consider for finalisation:

- in `test.yml` we now have **two** python 3.9 jobs (because ssh was removed). The only difference is the env IPP_ENABLE_CURVE variable. Do we need two job for this? Maybe we should switch one job to python 3.12 instead
- I have also added submitting the codecov report. I have no experience with that, so I'm not sure if this is working correctly

I have also reactivated all tests (hopefully) that were temporarly skipped/disabled